From 8f962a0ccbedd467b95a0aba4a9ef174261c3cd8 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Wed, 12 Sep 2018 12:32:40 +0530
Subject: [PATCH 0188/1051] soc: octeontx2: Make NDC enums reflect NDC hardware
 instance

Make NDC enums reflect NDC hardware instance.

Signed-off-by: Linu Cherian <lcherian@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu.c         |  6 ++--
 drivers/soc/marvell/octeontx2/rvu_debugfs.c | 15 ++++++----
 drivers/soc/marvell/octeontx2/rvu_struct.h  | 32 ++++++++++-----------
 3 files changed, 28 insertions(+), 25 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu.c b/drivers/soc/marvell/octeontx2/rvu.c
index 24fb20f1efa4..830078ee2ebb 100644
--- a/drivers/soc/marvell/octeontx2/rvu.c
+++ b/drivers/soc/marvell/octeontx2/rvu.c
@@ -407,9 +407,9 @@ static void rvu_reset_all_blocks(struct rvu *rvu)
 	rvu_block_reset(rvu, BLKADDR_SSO, SSO_AF_BLK_RST);
 	rvu_block_reset(rvu, BLKADDR_TIM, TIM_AF_BLK_RST);
 	rvu_block_reset(rvu, BLKADDR_CPT0, CPT_AF_BLK_RST);
-	rvu_block_reset(rvu, BLKADDR_NDC0, NDC_AF_BLK_RST);
-	rvu_block_reset(rvu, BLKADDR_NDC1, NDC_AF_BLK_RST);
-	rvu_block_reset(rvu, BLKADDR_NDC2, NDC_AF_BLK_RST);
+	rvu_block_reset(rvu, BLKADDR_NDC_NIX0_RX, NDC_AF_BLK_RST);
+	rvu_block_reset(rvu, BLKADDR_NDC_NIX0_TX, NDC_AF_BLK_RST);
+	rvu_block_reset(rvu, BLKADDR_NDC_NPA0, NDC_AF_BLK_RST);
 }
 
 static void rvu_scan_block(struct rvu *rvu, struct rvu_block *block)
diff --git a/drivers/soc/marvell/octeontx2/rvu_debugfs.c b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
index 10ae8f0ca7ed..a71fe3179aae 100644
--- a/drivers/soc/marvell/octeontx2/rvu_debugfs.c
+++ b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
@@ -546,7 +546,8 @@ static ssize_t rvu_dbg_npa_ndc_cache_display(struct file *filp,
 					     char __user *buffer,
 					     size_t count, loff_t *ppos)
 {
-	return ndc_blk_cache_stats(filp->private_data, NPA0_U, BLKADDR_NDC2);
+	return ndc_blk_cache_stats(filp->private_data, NPA0_U,
+				   BLKADDR_NDC_NPA0);
 }
 RVU_DEBUG_FOPS(npa_ndc_cache, npa_ndc_cache_display, NULL);
 
@@ -570,7 +571,7 @@ static ssize_t rvu_dbg_npa_ndc_hits_miss_display(struct file *filp,
 						 size_t count, loff_t *ppos)
 {
 	return ndc_blk_hits_miss_stats(filp->private_data,
-				      NPA0_U, BLKADDR_NDC2);
+				      NPA0_U, BLKADDR_NDC_NPA0);
 }
 RVU_DEBUG_FOPS(npa_ndc_hits_miss, npa_ndc_hits_miss_display, NULL);
 
@@ -1010,7 +1011,8 @@ static ssize_t rvu_dbg_nix_ndc_rx_cache_display(struct file *filp,
 						char __user *buffer,
 						size_t count, loff_t *ppos)
 {
-	return ndc_blk_cache_stats(filp->private_data, NIX0_RX, BLKADDR_NDC0);
+	return ndc_blk_cache_stats(filp->private_data, NIX0_RX,
+				   BLKADDR_NDC_NIX0_RX);
 }
 RVU_DEBUG_FOPS(nix_ndc_rx_cache, nix_ndc_rx_cache_display, NULL);
 
@@ -1018,7 +1020,8 @@ static ssize_t rvu_dbg_nix_ndc_tx_cache_display(struct file *filp,
 						char __user *buffer,
 						size_t count, loff_t *ppos)
 {
-	return ndc_blk_cache_stats(filp->private_data, NIX0_TX, BLKADDR_NDC1);
+	return ndc_blk_cache_stats(filp->private_data, NIX0_TX,
+				   BLKADDR_NDC_NIX0_TX);
 }
 RVU_DEBUG_FOPS(nix_ndc_tx_cache, nix_ndc_tx_cache_display, NULL);
 
@@ -1027,7 +1030,7 @@ static ssize_t rvu_dbg_nix_ndc_rx_hits_miss_display(struct file *filp,
 						    size_t count, loff_t *ppos)
 {
 	return ndc_blk_hits_miss_stats(filp->private_data,
-				      NPA0_U, BLKADDR_NDC0);
+				      NPA0_U, BLKADDR_NDC_NIX0_RX);
 }
 RVU_DEBUG_FOPS(nix_ndc_rx_hits_miss, nix_ndc_rx_hits_miss_display, NULL);
 
@@ -1036,7 +1039,7 @@ static ssize_t rvu_dbg_nix_ndc_tx_hits_miss_display(struct file *filp,
 						    size_t count, loff_t *ppos)
 {
 	return ndc_blk_hits_miss_stats(filp->private_data,
-				      NPA0_U, BLKADDR_NDC1);
+				      NPA0_U, BLKADDR_NDC_NIX0_TX);
 }
 RVU_DEBUG_FOPS(nix_ndc_tx_hits_miss, nix_ndc_tx_hits_miss_display, NULL);
 
diff --git a/drivers/soc/marvell/octeontx2/rvu_struct.h b/drivers/soc/marvell/octeontx2/rvu_struct.h
index 4d80c2595c35..ffb03e40b56b 100644
--- a/drivers/soc/marvell/octeontx2/rvu_struct.h
+++ b/drivers/soc/marvell/octeontx2/rvu_struct.h
@@ -15,22 +15,22 @@
  * RVU Block Address Enumeration
  */
 enum rvu_block_addr_e {
-	BLKADDR_RVUM    = 0x0ULL,
-	BLKADDR_LMT     = 0x1ULL,
-	BLKADDR_MSIX    = 0x2ULL,
-	BLKADDR_NPA     = 0x3ULL,
-	BLKADDR_NIX0    = 0x4ULL,
-	BLKADDR_NIX1    = 0x5ULL,
-	BLKADDR_NPC     = 0x6ULL,
-	BLKADDR_SSO     = 0x7ULL,
-	BLKADDR_SSOW    = 0x8ULL,
-	BLKADDR_TIM     = 0x9ULL,
-	BLKADDR_CPT0    = 0xaULL,
-	BLKADDR_CPT1    = 0xbULL,
-	BLKADDR_NDC0    = 0xcULL,
-	BLKADDR_NDC1    = 0xdULL,
-	BLKADDR_NDC2    = 0xeULL,
-	BLK_COUNT	= 0xfULL,
+	BLKADDR_RVUM		= 0x0ULL,
+	BLKADDR_LMT		= 0x1ULL,
+	BLKADDR_MSIX		= 0x2ULL,
+	BLKADDR_NPA		= 0x3ULL,
+	BLKADDR_NIX0		= 0x4ULL,
+	BLKADDR_NIX1		= 0x5ULL,
+	BLKADDR_NPC		= 0x6ULL,
+	BLKADDR_SSO		= 0x7ULL,
+	BLKADDR_SSOW		= 0x8ULL,
+	BLKADDR_TIM		= 0x9ULL,
+	BLKADDR_CPT0		= 0xaULL,
+	BLKADDR_CPT1		= 0xbULL,
+	BLKADDR_NDC_NIX0_RX	= 0xcULL,
+	BLKADDR_NDC_NIX0_TX	= 0xdULL,
+	BLKADDR_NDC_NPA0	= 0xeULL,
+	BLK_COUNT		= 0xfULL,
 };
 
 /*
-- 
2.17.1

